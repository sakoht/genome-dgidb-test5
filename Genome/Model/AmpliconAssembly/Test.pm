package Genome::Model::AmpliconAssembly::Test;

use strict;
use warnings;

use base 'Test::Class';

use File::Copy 'copy';
use File::Temp 'tempdir';
use Genome::InstrumentData::Sanger::Test;
use Genome::Model::InstrumentDataAssignment;
use Genome::ProcessingProfile::AmpliconAssembly::Test;
use Genome::Utility::FileSystem;
use Test::More;

#< DIR >#
sub test_dir {
    return '/gsc/var/cache/testsuite/data/Genome-Model-AmpliconAssembly';
}

sub _test_dir_for_type {
    die "No type to get test dir\n" if not defined $_[0] or $_[0] eq __PACKAGE__;
    return test_dir().'/'.$_[0];
}

#< MOCK ># 
sub create_mock_model {
    my $self = shift;

    # Processing profile
    my $pp = Genome::ProcessingProfile::AmpliconAssembly::Test->create_mock_processing_profile
        or return;

    #my $data_dir = File::Temp::tempdir(DIR => $self->test_dir, CLEANUP => 0);
    my $data_dir = File::Temp::tempdir(CLEANUP => 1);
    die "Can't make temp data directory for mock model\n" unless -d $data_dir;

    # Model
    my $model = Genome::Model::AmpliconAssembly->create_mock(
        id => -5000,
        genome_model_id => -5000,
        name => 'mr. mock',
        subject_name => 'mock_dna',
        subject_type => 'dna_resource_item_name',
        processing_profile_id => $pp->id,
        processing_profile => $pp,
        data_directory => $data_dir,
    )
        or die "Can't create mock model for amplicon assembly\n";
    for my $pp_param ( Genome::ProcessingProfile::AmpliconAssembly->params_for_class ) {
        $model->set_always($pp_param, $pp->$pp_param);
    }
    for my $dir_type ( Genome::Consed::Directory->directories ) {
        my $dir = $data_dir.'/'.$dir_type; 
        $model->set_always($dir_type, $dir);
        mkdir $dir
            or die "Can't make directory ($dir): $!\n";
    }
    $model->mock(
        'amplicons',
        sub{  
            my $dh = Genome::Utility::FileSystem->open_directory( $model->chromat_dir )
                or return;

            my %amplicons;
            while ( my $scf = $dh->read ) {
                next if $scf =~ m#^\.#;
                $scf =~ s#\.gz##;
                $scf =~ /^(.+)\.[bg]\d+$/
                    or next;
                push @{$amplicons{$1}}, $scf;
            }
            $dh->close;

            return \%amplicons;
        },
    );

    #< FILES >#
    for my $type (qw/ assembly reads processed assembly.confirmed assembly.unconfirmed /) {
        my $method = $type.'_fasta';
        $model->set_always($method, sprintf('%s/%s.%s.fasta', $data_dir, $model->subject_name, $type));
    }
    $model->set_always('metrics_file',  sprintf('%s/%s.metrics.txt', $data_dir, $model->subject_name));

    return $model;
}

sub create_mock_model_with_instrument_data {
    my $self = shift;

    my $model = $self->create_mock_model
        or return;

    my $inst_data = Genome::InstrumentData::Sanger::Test->create_mock_instrument_data; # this dies if no workee
    my $ida = Genome::Model::InstrumentDataAssignment->create_mock(
        id => -5000,
        model_id => $model->id,
        instrument_data_id => $inst_data->id,
        first_build_id => undef,
    )
        or die "Can't create mock instrument data assignment\n";
    $ida->set_always('model_id', $model);
    $ida->set_always('instrument_data', $inst_data);
    $ida->mock(
        'first_build_id', sub { 
            my ($ida, $fbi) = @_;
            $ida->{first_build_id} = $fbi if defined $fbi;
            return $ida->{first_build_id}; 
        },
    );
    $model->set_always('instrument_data', ( $inst_data ));
    $model->set_always('instrument_data_assignments', ( $ida ));

    return $model;
}

#< COPY DATA >#
sub copy_dirs_to_model {
    my ($self, $model, @dirs) = @_;

    die "Need dir types to copy from\n" unless @dirs;

    for my $dir ( @dirs ) {
        my $dest = ( $model->can($dir) )
        ? $model->$dir
        : $model->data_directory;
        $self->_copy_dir(_test_dir_for_type($dir), $dest)
            or return;
    }

    return 1;
}

sub _copy_dir {
    my ($self, $source, $dest) = @_;

    my $dh = Genome::Utility::FileSystem->open_directory($source)
        or return;

    while ( my $file = $dh->read ) {
        next if $file =~ m#^\.#;
        File::Copy::copy("$source/$file", $dest)
            or die "Can't copy ($source/$file) to ($dest): $!\n";
    }

    return 1;

}

#< REAL MODEL FOR TESTING >#

1;

=pod

=head1 Name

ModuleTemplate

=head1 Synopsis

=head1 Usage

=head1 Methods

=head2 

=over

=item I<Synopsis>

=item I<Arguments>

=item I<Returns>

=back

=head1 See Also

=head1 Disclaimer

Copyright (C) 2005 - 2008 Washington University Genome Sequencing Center

This module is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY or the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for more details.

=head1 Author(s)

B<Eddie Belter> I<ebelter@watson.wustl.edu>

=cut

#$HeadURL$
#$Id$

