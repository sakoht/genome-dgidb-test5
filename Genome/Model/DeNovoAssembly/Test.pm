package Genome::Model::DeNovoAssembly::Test;

use strict;
use warnings;

require Carp;
use Data::Dumper 'Dumper';
require Genome::Model::Test;
require Genome::Utility::TestBase;

BEGIN {
    $ENV{UR_USE_DUMMY_AUTOGENERATED_IDS} = 1;
    $ENV{UR_DBI_NO_COMMIT} = 1;
}

sub processing_profile_params_for_assembler_and_platform {
    my ($self, %params) = @_;

    my $sequencing_platform = delete $params{sequencing_platform};
    Carp::confess "No sequencing platform given to create mock processing profile" unless $sequencing_platform;
    my $assembler_name = delete $params{assembler_name};
    Carp::confess "No assembler name given to create mock processing profile" unless $assembler_name;
    Carp::confess("Unknown params to get mock processing profile\n".Dumper(\%params)) if %params;

    my %assembler_sequencing_paltorm_params = (
        velvet_solexa =>  { 
            read_coverage => 0.5,#25000,
            assembler_version => '0.7.57-64',
            assembler_params => '-hash_sizes 31 33 35 -ins_length 260',
            read_trimmer_name => 'basic',
            read_trimmer_params => '-trim_length 10',
        },
        newbler_454 => {
        },
    );

    my $specific_params = $assembler_sequencing_paltorm_params{ $assembler_name.'_'.$sequencing_platform };
    unless ( $specific_params ) {
        Carp::confess "Invalid assembler ($assembler_name) and sequencing platform ($sequencing_platform) combination";
    }

    $specific_params->{name} = 'De Novo Assembly Test';
    $specific_params->{sequencing_platform} = $sequencing_platform;
    $specific_params->{assembler_name} = $assembler_name;
    
    return %$specific_params;
}

sub get_mock_processing_profile {
    my $self = shift;

    my %params = $self->processing_profile_params_for_assembler_and_platform(@_) or Carp::confess;
    my $pp = Genome::Model::Test->get_mock_processing_profile(
        class => 'Genome::ProcessingProfile::DeNovoAssembly',
        type_name => 'de novo assembly',
        %params,
    ) or Carp::confess "Can't get mock de novo assembly processing profile";
    
    Genome::Utility::TestBase->mock_methods(
        $pp,
        (qw/ 
            _name_for_operation

            _class_for_operation
            class_for_assembler
            class_for_read_filter
            class_for_read_trimmer

            _params_as_hash_for_operation
            assembler_params_as_hash
            read_trimmer_params_as_hash
            read_filter_params_as_hash
            
            _object_for_operation
            create_read_filter
            create_read_trimmer
            
            _validate_assembler_and_params
            _validate_operation_and_params
            _validate_read_filter_and_params
            _validate_read_trimmer_and_params
            
            status_message
            /),
    );

    return $pp;
}

sub get_mock_subject {
    my $self = shift;

    my $sample_name = 'H_KT-185-1-0089515594';
    
    # 2851686380
    my $taxon = Genome::Utility::TestBase->create_mock_object(
        class => 'Genome::Taxon',
        domain => 'Bacteria',
        species_name => 'Escherichia coli 185-1',
        current_default_org_prefix => undef,
        estimated_genome_size => 4500000,,
        current_genome_refseq_id => undef,
        ncbi_taxon_id => undef,
        ncbi_taxon_species_name => undef,
        species_latin_name => 'Escherichia coli',
        strain_name => 185-1,
        subject_type => 'organism taxon',
    ) or Carp::confess "Can't create mock taxon";

    # 2851686381
    my $source = Genome::Utility::TestBase->create_mock_object(
        class => 'Genome::SampleSource',
        id => 2851686381,
        taxon_id => $taxon->id,
        name => $sample_name,
    ) or Carp::confess "Can't create mock source";

    # 2851686382
    my $subject = Genome::Utility::TestBase->create_mock_object(
        class => 'Genome::Sample',
        id => 2851686382,
        source_id => $source->id,
        source_type => 'organism individual',
        name => 'H_KT-185-1-0089515594',
        common_name => undef,
        extraction_label => '0089515594',
        extraction_type => 'genomic dna',
        extraction_desc => undef,
        cell_type => 'primary',
        tissue_desc => undef,
        tissue_label => undef,
        organ_name => undef,
        taxon_id => $taxon->id,
    ) or Carp::confess "Can't create mock sample";

    return $subject;
}

sub get_mock_model {
    my $self = shift;

    # pp
    my $pp = $self->get_mock_processing_profile(@_) 
        or Carp::confess("Can't create mock de novo assembly pp");

    # subject
    my $subject = $self->get_mock_subject
        or Carp::confess("Can't get mock subject for de novo assembly model");

    # model
    my $model = Genome::Model::Test->get_mock_model(
        class => 'Genome::Model::DeNovoAssembly',
        processing_profile => $pp,
        subject => $subject,
    ) or Carp::confess "Can't get mock de novo assembly model";

    # inst data
    my $sequencing_platform = $pp->sequencing_platform;
    my $inst_data_method = '_get_mock_'.$sequencing_platform.'_instrument_data';
    my $inst_data = $self->$inst_data_method
        or Carp::confess "Can't create mock instrument data for de novo assembly model";
    my $instrument_data_assignment = Genome::Utility::TestBase->create_mock_object(
        class => 'Genome::Model::InstrumentDataAssignment',
        model => $model,
        model_id => $model->id,
        instrument_data => $inst_data,
        instrument_data_id => $inst_data->id,
        first_build_id => undef,
    ) or Carp::confess("Can't assign instrument data to de novo assembly model");

    return $model;
}

sub get_mock_build {
    my ($self, %params) = @_;

    my $model = delete $params{model};
    Carp::confess "No mc116s model given to create mock build" unless $model;
    my $use_example_directory = delete $params{use_example_directory};
    Carp::confess("Unknown params to get mock build:\n".Dumper(\%params)) if %params;
    
    my $build = Genome::Model::Test->get_mock_build(
        class => 'Genome::Model::Build::DeNovoAssembly::'.Genome::Utility::Text::string_to_camel_case($model->assembler_name),
        model => $model,
        data_directory => ( 
            $use_example_directory
            ? $self->example_directory_for_model($model)
            : undef 
        ),
    )
        or Carp::confess("Can't add mock build to model");

    $build->mock('instrument_data', sub{ return $model->instrument_data; });

    Genome::Utility::TestBase->mock_methods(
        $build,
        (qw/
            description

            interesting_metric_names
            set_metrics

            genome_size
            estimate_average_read_length
            calculate_read_limit_from_read_coverage

            edit_dir
            
            collated_fastq_file
            assembly_afg_file
            sequences_file
            velvet_ace_file
            contigs_fasta_file
            assembly_afg_file
            contigs_fasta_file
            sequences_file
            ace_file
            gap_file
            contigs_bases_file
            contigs_quals_file
            read_info_file
            reads_placed_file
            supercontigs_agp_file
            supercontigs_fasta_file
            stats_file
            status_message

            /)
    );

    Genome::Utility::TestBase->mock_methods(
        $build,
        map { join('_', split(m#\s#)) } $build->interesting_metric_names,
    );

    return $build;
}

#< Example Files >#
sub base_directory {
    return '/gsc/var/cache/testsuite/data/Genome-Model/DeNovoAssembly';
}

sub example_directory_for_model {
    my ($self, $model) = @_;

    Carp::confess "No model to get example directory" unless $model;
    
    my $dir = $self->base_directory.'/'.$model->assembler_name.'_'.$model->sequencing_platform.'_build';
    Carp::confess("Example directory ($dir) for de novo assembly model does not exist.") unless -d $dir;
    
    return $dir;
}

sub example_fastq_file_for_model {
    my ($self, $pp) = @_;

    my $dir = $self->example_directory_for_model($pp);

    return $dir.'/collated.fastq';
}

sub example_assembly_afg_file_for_model {
    my ($self, $pp) = @_;

    my $dir = $self->example_directory_for_model($pp);

    return $dir.'/velvet_asm.afg';
}

sub example_sequences_file_for_model {
    my ($self, $pp) = @_;

    my $dir = $self->example_directory_for_model($pp);

    return $dir.'/Sequences';
}

sub example_contigs_fasta_file_for_model {
    my ($self, $pp) = @_;

    my $dir = $self->example_directory_for_model($pp);

    return $dir.'/contigs.fa';
}

sub example_stats_file_for_model {
    my ($self, $model) = @_;

    my $dir = $self->example_directory_for_model($model);

    return $dir.'/edit_dir/stats.txt';
}

#<>#

#< Instrument Data >#
sub _instrument_data_dir {
    return $_[0]->base_directory.'/inst_data/';
}

sub _get_mock_solexa_instrument_data {
    my $self = shift;

    my $id = 2854709902;
    my $inst_data = Genome::Utility::TestBase->create_mock_object(
        class => 'Genome::InstrumentData::Solexa',
        id => $id,
        seq_id => $id,
        sequencing_platform => 'solexa',
        flow_cell_id => '61H93',
        lane => 8,
        read_length => 100,
        archive_path => $self->_instrument_data_dir.'/'.$id.'/sequence_61H93_8_CGATGT.tar.gz',
        index_sequence => 'CGATGT',
        subset_name => '8-CGATGT',
        is_external => 0,
        is_paired_end => 1,
        library_id => '2852968107',
        library_name => 'H_KT-185-1-0089515594-lib1',
        sample_name => 'H_KT-185-1-0089515594',
    ) or Carp::confess "Can't create mock solexa instrument data";

    return $inst_data;
}

sub _get_mock_454_instrument_data {
    # FIXME!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
    my $self = shift;

    my $id = 222;
    my $full_path = $self->_instrument_data_dir('454').'/'.$id;
    #Carp::confess "Mock instrument data directory ($full_path) does not exist" unless -d $full_path;
    my $inst_data = Genome::Utility::TestBase->create_mock_object(
        class => 'Genome::InstrumentData::454',
        id => $id,
        seq_id => $id,
        region_id => $id,
        analysis_name => 'D_2010_01_10_04_22_16_blade9-2-5_fullProcessing',
        region_number => 2,
        total_reads => 20,
        total_raw_wells => 1176187,
        total_key_pass => 1169840,
        incoming_dna_name => 'Pooled_Library-2009-12-31_1-1',
        copies_per_bead => 2.5,
        run_name => 'R_2010_01_09_11_08_12_FLX08080418_Administrator_100737113',
        key_pass_wells => 1170328,
        predicted_recovery_beads => 371174080,
        fc_id => undef,
        sample_set => 'Tarr NEC 16S Metagenomic Sequencing master set',
        research_project => 'Tarr NEC 16S Metagenomic Sequencing',
        paired_end => 0,
        sample_name => 'H_MA-.0036.01-89503877',
        library_name => 'Pooled_Library-2009-12-31_1',
        beads_loaded => 1999596,
        ss_id => undef,
        supernatant_beads => 254520,
        sample_id => 2847037746,
        library_id => 2848636935,
        #library_name => 'Pooled_DNA-2009-03-09_23-lib1',
        sequencing_platform => '454',
        full_path => $full_path,
    ) or Carp::confess "Unable to create mock 454 id #";
    $inst_data->mock('fasta_file', sub { 
            return $full_path.'/zymo4_pool7_DNA_extract.fasta';
        }
    );
    $inst_data->mock('dump_to_file_system', sub{ return 1; });

    return $inst_data;
}
#<>#

1;

=pod

=head1 Tests

=head1 Disclaimer

 Copyright (C) 2010 Washington University Genome Sequencing Center

 This script is distributed in the hope that it will be useful,
 but WITHOUT ANY WARRANTY or the implied warranty of MERCHANTABILITY
 or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public
 License for more details.

=head1 Author(s)

 Eddie Belter <ebelter@genome.wustl.edu>

=cut

#$HeadURL: svn+ssh://svn/srv/svn/gscpan/perl_modules/trunk/Genome/MetagenomicComposition16s/Test.pm $
#$Id: Test.pm 54265 2010-01-05 16:50:07Z ebelter $

