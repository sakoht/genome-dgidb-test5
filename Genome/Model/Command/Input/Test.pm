
###########################################################################

package Genome::Model::Command::Input::Test;

use strict;
use warnings;

use base 'Test::Class';

use Data::Dumper 'Dumper';
use Test::More;

sub test_class {
    return 'Genome::Model::Command::Input';
}



###########################################################################

package Genome::Model::Command::Input::TestBase;

use strict;
use warnings;

use base 'Genome::Utility::TestCommandBase';

use Carp 'confess';
use Data::Dumper 'Dumper';
use Test::More;
require Genome::Model::Test;
require File::Temp;
require File::Path;

sub valid_param_sets {
    return grep { $_->{model_identifier} = $_[0]->_model->id } $_[0]->_valid_param_sets;
}

sub invalid_param_sets {
    return map { $_->{model_identifier} = $_[0]->_model->id } $_[0]->_valid_param_sets;
}

sub _model {
    my $self = shift;

    unless ( $self->{_model} ) {
        $self->{_model} = Genome::Model::Test->create_basic_mock_model(type_name => 'tester')
            or confess "Can't create mock tester model.";
    }

    return $self->{_model};
}

sub startup : Tests(startup => 1) {
    my $self = shift;
    
    no warnings 'once';
    *Genome::Model::Command::get_model_type_names = sub{ 
        return (qw/ tester /); 
    };
    is_deeply(
        [ Genome::Model::Command->get_model_type_names ],
        [qw/ tester /],
        'Set tester type name.',
    );

    $self->_startup or confess "Failed startup.";
    
    return 1;
}

sub _startup {
    return 1;
}

# 
sub __pre_execute {
    my $self = shift;

    print Dumper([$self->_model->inputs]);
    
    return 1;
}

sub __post_execute {
    my $self = shift;

    print Dumper([$self->_model->inputs]);
    
    return 1;
}

###########################################################################

package Genome::Model::Command::Input::Add::Test;

use strict;
use warnings;

use base 'Genome::Model::Command::Input::TestBase';

use Data::Dumper 'Dumper';
use Test::More;

sub test_class {
    return 'Genome::Model::Command::Input::Add';
}

sub _valid_param_sets {
    return (
        { # add multiple 'value_ids' 
            name => 'friends',
            values => 'Watson,Crick',
        },
        { # add single 'value'
            name => 'inst_data',
            values => '2sep09.934pmaa1',
        },
    );
}

sub invalid_param_sets {
    return (
        {# try to add a not 'is_many' input
            name => 'coolness', 
            values => 'none',
        },
        { # try to add Crick again
            name => 'friends',
            values => 'Crick',
        },
        { # try to add id that don't exist
            name => 'inst_data',
            values => '2sep09.934noexist',
        },
    );
}

sub _pre_execute {
    my ($self, $obj) = @_;

    my $name = $obj->name;
    my @values = $self->_model->$name;
    ok(!@values, "No $name found in model.");
    
    return 1;
}

sub _post_execute {
    my ($self, $obj) = @_;

    my $name = $obj->name;
    my @values = $self->_model->$name;
    ok(@values, "Added $name in model.");
    
    return 1;
}

###########################################################################

package Genome::Model::Command::Input::Remove::Test;

use strict;
use warnings;

use base 'Genome::Model::Command::Input::TestBase';

use Carp 'confess';
use Data::Dumper 'Dumper';
use Test::More;

sub test_class {
    return 'Genome::Model::Command::Input::Remove';
}

sub _valid_param_sets {
    return (
        { # remove multiple 'value_ids' 
            name => 'friends',
            values => 'Watson,Crick',
        },
        { # remove single 'value'
            name => 'inst_data',
            values => '2sep09.934pmaa1',
        },
    );
}

sub invalid_param_sets {
    return (
        {# try to remove a not 'is_many' input
            name => 'coolness', 
            values => 'none',
        },
        { # try to remove input that is not linked to the model anymore
            name => 'inst_data',
            values => '2sep09.934pmaa1',
        },
    );
}

sub friends {
    return 'Crick';
}

sub inst_data {
    return Genome::InstrumentData::Sanger->get('2sep09.934pmaa1')
}

sub _startup {
    my $self = shift;

    my $model = $self->_model;

    $model->add_friend('Crick');
    $model->add_friend('Watson');
    my @friends = $model->friends();
    is_deeply(\@friends, [qw/ Crick Watson /], 'Added friends to remove.');
    my $id = Genome::InstrumentData::Sanger->get('2sep09.934pmaa1')
        or confess "Can't get sange id 2sep09.934pmaa1";
    $model->add_inst_data($id);
    my @inst_data = $model->inst_data;
    is_deeply(\@inst_data, [ $id ], 'Added instr_data to remove');

    return 1;
}

sub _post_execute {
    my ($self, $obj) = @_;

    my $name = $obj->name;
    my @values = $self->_model->$name;
    #print Dumper(\@values);
    ok(!@values, "Removed $name from model.");
    
    return 1;
}

###########################################################################

package Genome::Model::Command::Input::List::Test;

use strict;
use warnings;

use base 'Genome::Model::Command::Input::TestBase';

use Data::Dumper 'Dumper';
use Test::More;

sub test_class {
    return 'Genome::Model::Command::Input::List';
}

sub valid_param_sets {
    return (
        {
            filter => 'model_id='.$_[0]->_model->id,
        },
    );
}

sub invalid_param_sets {
}

sub _startup {
    my $self = shift;

    $self->_model->coolness('low');
    is($self->_model->coolness, 'low', 'Set coolness (input)');

    return 1;
}

###########################################################################

package Genome::Model::Command::Input::Names::Test;

use strict;
use warnings;

use base 'Genome::Utility::TestCommandBase';

use Data::Dumper 'Dumper';
use Test::More;

sub test_class {
    return 'Genome::Model::Command::Input::Names';
}

sub valid_param_sets {
    return (
        {
            type_name => 'tester',
        },
    );
}

sub invalid_param_sets {
    return (
        {
            type_name => 'none',
        },
    );
}

sub startup : Tests(startup) {
    my $self = shift;
    
    no warnings 'once';
    no warnings 'redefine';
    *Genome::Model::Command::get_model_type_names = sub{ 
        return (qw/ tester /); 
    };
    
    return 1;
}

###########################################################################

package Genome::Model::Command::Input::Update::Test;

use strict;
use warnings;

use base 'Genome::Model::Command::Input::TestBase';

use Carp 'confess';
use Data::Dumper 'Dumper';
use Test::More;

sub test_class {
    return 'Genome::Model::Command::Input::Update';
}

sub _valid_param_sets {
    return (
        { # remove single 'value'
            name => 'coolness',
            value => 'moderate',
        },
        { # remove single 'value'
            name => 'coolness',
            value => 'UNDEF',
        },
    );
}

sub invalid_param_sets {
    return (
        { # try to update an 'is_many' input
            name => 'friends',
            value => 'Watson',
        },
    );
}

###########################################################################

1;

=pod

=head1 Tests

=head1 Disclaimer

 Copyright (C) 2009 Washington University Genome Sequencing Center

 This script is distributed in the hope that it will be useful,
 but WITHOUT ANY WARRANTY or the implied warranty of MERCHANTABILITY
 or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public
 License for more details.

=head1 Author(s)

 Eddie Belter <ebelter@watson.wustl.edu>

=cut

#$HeadURL$
#$Id$

