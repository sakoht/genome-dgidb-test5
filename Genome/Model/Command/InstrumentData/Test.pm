
########################################################################

package Genome::Model::Command::InstrumentData::Test;

use strict;
use warnings;

#use base 'Genome::Utility::TestBase';


########################################################################

package Genome::Model::Command::InstrumentData::Assign::Test;

use strict;
use warnings;

use base 'Genome::Utility::TestBase';

require Genome::Model::Test;
use Test::More;

sub test_class {
    return 'Genome::Model::Command::InstrumentData::Assign';
}

sub params_for_test_class {
    return (
        model_id => $_[0]->model_id,
    );
}

sub required_params_for_class {
    return (qw/ model_id /);
}

sub model { return $_[0]->{_model}; }
sub model_id { return $_[0]->{_model}->id; }
sub instrument_data { return $_[0]->{_instrument_data}; }

sub create_mock_model_and_instrument_data : Test(startup => 2) {
    my $self = shift;

    # model
    my $model = Genome::Model::Test->create_basic_mock_model(type_name => 'tester');
    ok($model, 'Created mock model') or die;
    $self->{_model} = $model;

    # instrument data
    my @instrument_data = Genome::Model::Test->create_mock_sanger_instrument_data(4);
    is(scalar(@instrument_data), 4, 'Created 4 instrument data');
    $self->{_instrument_data} = \@instrument_data;

    $model->set_list('compatible_instrument_data', @instrument_data);
    
    return 1;
}

sub test_01_list : Tests {
    my $self = shift;

    ok($self->{_object}->execute, 'List instrument data');

    return 1;
}

sub test_02_assign_single_id : Tests {
    my $self = shift;

    my $assign = $self->test_class->create(
        $self->params_for_test_class,
        instrument_data_id => $self->instrument_data->[0]->id,
    );
    ok($assign->execute, 'Assign single instrument data');

    return 1;
}

sub test_03_assign_multiple_id : Tests {
    my $self = shift;

    my $assign = $self->test_class->create(
        $self->params_for_test_class,
        instrument_data_ids => join(
            ' ',
            $self->instrument_data->[1]->id,
            $self->instrument_data->[2]->id,
        ),
    );
    ok($assign->execute, 'Assign multiple instrument data');

    return 1;
}

sub test_04_assign_all : Tests {
    my $self = shift;

    my $assign = $self->test_class->create(
        $self->params_for_test_class,
        all => 1,
    );
    ok($assign->execute, 'Assign all instrument data');

    return 1;
}

sub test_05_multiple_actions : Tests {
    my $self = shift;

    my $assign = $self->test_class->create(
        all => 1,
        instrument_data_ids => '44 44',
    );
    ok(!$assign, 'Failed as expected - request multiple actions');

    return 1;
}

########################################################################

1;

#$HeadURL$
#$Id$
