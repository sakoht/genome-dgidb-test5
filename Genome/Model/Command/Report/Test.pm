package Genome::Model::Command::Report::TestBase;

use strict;
use warnings;

use Carp 'confess';
$SIG{__DIE__} = sub{ confess(@_) };

use base 'Genome::Utility::TestBase';

use Data::Dumper 'Dumper';
use Genome;
use Genome::Model::AmpliconAssembly::Test;
use Test::More;

sub _model {
    my $self = shift;

    unless ( $self->{_model} ) {
        $self->{_model} = Genome::Model::AmpliconAssembly::Test->create_mock_model(use_test_dir => 1)
            or die "Can't create mock model\n";
    }

    return $self->{_model};
}

sub _build {
    my $self = shift;

    return $self->_model->latest_complete_build;
}

sub params_for_test_class {
    return (
        build_id => $_[0]->_build->build_id,
        $_[0]->_params_for_test_class,
    );
}

sub _params_for_test_class {
    return;
}

sub test_01_execute : Tests {
    my $self = shift;

    ok($self->{_object}->execute, 'Execute');
    
    return 1;
}

########################################################

package Genome::Model::Command::Report::GenerateTest;

use strict;
use warnings;

use base 'Genome::Model::Command::Report::TestBase';

use Data::Dumper 'Dumper';

sub test_class {
    return 'Genome::Model::Command::Report::Generate';
}

sub _params_for_test_class {
    return (
        report_name => 'Assembly Stats',
        directory => '/gscuser/ebelter/Desktop',
        force => 1,
    );
}

########################################################

package Genome::Model::Command::Report::GetDatasetTest;

use strict;
use warnings;

use base 'Genome::Model::Command::Report::TestBase';

use Data::Dumper 'Dumper';

sub test_class {
    return 'Genome::Model::Command::Report::GetDataset';
}

sub _params_for_test_class {
    return (
        report_name => 'Assembly Stats',
        dataset_name => 'stats',
        output_type => 'csv',
        #output_type => 'xml',
    );
}

1;

=pod

=head1 Name

ModuleTemplate

=head1 Synopsis

=head1 Usage

=head1 Methods

=head2 

=over

=item I<Synopsis>

=item I<Arguments>

=item I<Returns>

=back

=head1 See Also

=head1 Disclaimer

Copyright (C) 2005 - 2009 Genome Center at Washington University in St. Louis

This module is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY or the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for more details.

=head1 Author(s)

B<Eddie Belter> I<ebelter@genome.wustl.edu>

=cut

#$HeadURL$
#$Id$

