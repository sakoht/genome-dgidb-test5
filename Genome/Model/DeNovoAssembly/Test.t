#!/gsc/bin/perl

use strict;
use warnings;

use above "Genome";

use Data::Dumper 'Dumper';
use Test::More;

use_ok('Genome::Model::DeNovoAssembly::Test') or die;

my $rv;

#< PP >#
# fail - no seq platform
eval{
    $rv = Genome::Model::DeNovoAssembly::Test->get_mock_model();
};
diag($@);
ok(
    !$rv && $@,
    'Failed as expected - get w/o seq platform',
);
# fail - no assembler
eval{
    $rv = Genome::Model::DeNovoAssembly::Test->get_mock_model(
        sequencing_platform => 'solexa',
    );
};
diag($@);
ok(
    !$rv && $@,
    'Failed as expected - create w/o assembler',
);
# fail - invalid seq platform/assembler combo
eval{
    $rv = Genome::Model::DeNovoAssembly::Test->get_mock_model(
        assembler_name => 'consed',
        sequencing_platform => 'solexa',
    );
};
diag($@);
ok(
    !$rv && $@,
    'Failed as expected - create w/ invalid seq paltform/assembler combo',
);

# valid 
my $model = Genome::Model::DeNovoAssembly::Test->get_mock_model(
    assembler_name => 'velvet',
    sequencing_platform => 'solexa',
);
ok($model, 'Got mock DNA model') or die;
my @inst_data = $model->instrument_data;
ok(@inst_data, 'Model instrument data');
ok(-s $inst_data[0]->archive_path, 'Solexa archive path exists');

done_testing();
exit;

#$HeadURL$
#$Id$
