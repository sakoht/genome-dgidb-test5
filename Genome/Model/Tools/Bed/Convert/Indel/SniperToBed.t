#!/gsc/bin/perl

use strict;
use warnings;


use Test::More tests => 4;

use above 'Genome';

use_ok('Genome::Model::Tools::Bed::Convert::Indel::SniperToBed');

my $tmpdir = File::Temp::tempdir('Bed-Convert-Indel-SniperToBedXXXXX', DIR => '/gsc/var/cache/testsuite/running_testsuites/', CLEANUP => 1);
my $output_file = join('/', $tmpdir, 'output');

my $input_file = __FILE__ . '.input';
my $expected_file = __FILE__ . '.expected';

my $command = Genome::Model::Tools::Bed::Convert::Indel::SniperToBed->create( source => $input_file, output => $output_file );
ok($command, 'Command created');
my $rv = $command->execute;
ok($rv, 'Command completed successfully');

my $diff = Genome::Utility::FileSystem->diff_file_vs_file($output_file, $expected_file);
ok(!$diff, 'output matched expected result')
    or diag("diff results:\n" . $diff);
