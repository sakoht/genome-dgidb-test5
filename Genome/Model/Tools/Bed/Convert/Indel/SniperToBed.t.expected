1	5	7	*/A
1	6	8	*/A
1	8	10	*/C
1	10	12	*/N
1	12	13	A/*
1	14	16	*/C
1	17	19	*/T
1	18	20	*/A
1	20	22	*/C
1	23	25	*/T
1	26	28	*/C
1	27	29	*/CCAAA
1	30	31	A/*
1	32	34	*/C
1	34	37	CTA/*
1	35	36	T/*
1	36	37	A/*
1	37	39	*/ACC
1	38	40	*/C
1	39	41	*/A
